packages/addresses/src/base58.ts

```diff
@@ -52,3 +52,18 @@ export function getBase58EncodedAddressComparator(): (x: string, y: string) => n
         usage: 'sort',
     }).compare;
 }
+
+export async function createBase58AddressWithSeed(
+    baseAddress: Base58EncodedAddress,
+    seed: string,
+    programAddress: Base58EncodedAddress
+): Promise<Base58EncodedAddress> {
+    const { serialize, deserialize } = getBase58EncodedAddressCodec();
+    const seedBytes = new TextEncoder().encode(seed);
```

Comment:
I was going to point out that `TextEncoder` isn't available in React Native, but:

* We also use it in `program-derived-address.ts`
* We should presume that, in React Native, we need to call out to an async API to do this encoding, but you've already made this method `async` so… carry on.